### ISSUE TITLE
.pth files are not processed when they are symbolic links

### ISSUE THREAD

**Issue by ipkiss42** [microsoft/pyright#1582]
**Describe the bug**
An import for a module accessible in the venv "via a .pth" does not work when the .pth file is a symbolic link (it works when it is a regular file).

**To Reproduce**

Create the following structure in `/tmp/pyright1`:

```
/tmp/pyright1
├── python3.8
│   ├── bin
│   │   └── ...
│   └── ...
└── script.sh
```

The `python3.8` directory contains a regular Python 3.8 venv.

`script.sh`:

```sh
#!/bin/sh
VENV_PATH="$1"
VENV="$2"

PWD=$(pwd)

rm -rf src && mkdir -p src
echo "import bar" > src/pth_bug.py
cat > pyrightconfig.json << EOF
{
    "venvPath": "$VENV_PATH",
    "venv": "$VENV",
    "executionEnvironments": [
        {
            "root": "src"
        }
    ]
}
EOF

rm -f foo.pth && echo "foo" > foo.pth
mkdir -p $VENV_PATH/$VENV/lib/python3.8/site-packages/foo
touch $VENV_PATH/$VENV/lib/python3.8/site-packages/foo/bar.py
ln -sf $PWD/foo.pth $VENV_PATH/$VENV/lib/python3.8/site-packages/foo.pth

export PATH=/tmp/node_modules/.bin:$PATH
(cd src && pyright --version && pyright --verbose)
```

Then, from `/tmp/pyright1` run `sh ./script.sh . python3.8`

**Expected behavior**
No error

**Screenshots or Code**
Pyright output:
```
pyright 1.1.118
Loading configuration file at /tmp/pyright1/pyrightconfig.json
Assuming Python version 3.8
Assuming Python platform Linux
No include entries specified; assuming /tmp/pyright1
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /tmp/pyright1/typings is not a valid directory.
Searching for source files
Found 7 source files
Could not import 'bar' in file '/tmp/pyright1/src/pth_bug.py'
  Looking for typeshed stdlib path
  Attempting to resolve using root path '/tmp/node_modules/pyright/dist/typeshed-fallback/stdlib'
  Typeshed path not found
  Looking in stubPath '/tmp/pyright1/typings'
  Attempting to resolve using root path '/tmp/pyright1/typings'
  Attempting to resolve using root path '/tmp/pyright1/typings'
  Looking in root directory of execution environment '/tmp/pyright1/src'
  Attempting to resolve using root path '/tmp/pyright1/src'
  Attempting to resolve using root path '/tmp/pyright1/src'
  Looking in python search path '/tmp/pyright1/python3.8/lib/python3.8/site-packages'
  Attempting to resolve using root path '/tmp/pyright1/python3.8/lib/python3.8/site-packages'
  Attempting to resolve using root path '/tmp/pyright1/python3.8/lib/python3.8/site-packages'
  Looking for typeshed path
  Looking for typeshed stubs path
  Typeshed path not found
/tmp/pyright1/src/pth_bug.py
  /tmp/pyright1/src/pth_bug.py:1:8 - error: Import "bar" could not be resolved (reportMissingImports)
1 error, 0 warnings, 0 infos
Completed in 0.513sec
```

**VS Code extension or command-line**
Version 1.1118 (CLI)

**Comment by erictraut** [comment]
I don't see anywhere in the Python spec that indicates that ".pth" files can be symlinks. [This documentation](https://docs.python.org/3/library/site.html) seems to be clear that they are always text files. Can you point me to any documentation that says symlinks are officially supported?

**Comment by ipkiss42** [comment]
The documentation says neither that symlinks are supported nor not supported: it simply doesn't call them out explicitly, as if they do not deserve a special handling.

The current implementation also treats symlinks the same as regular files, both [when listing them](https://github.com/python/cpython/blob/3.9/Lib/site.py#L202-L206) and when [processing them](https://github.com/python/cpython/blob/3.9/Lib/site.py#L158-L162). They do work fine with Python itself.

**Comment by erictraut** [comment]
So this feature request isn't really specific to ".pth" files. It's a general request for supporting symlinks during import resolution.

**Comment by ipkiss42** [comment]
Possibly, I'm not sure. I have had no other issue with symlinks so far, so I don't know what else than ".pth" files might be missing.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.120, which I just published. It will also be included in the next release of pylance.
